MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -42,6 +44,28 @@ final class MapViewModel: ObservableObject {
         CLLocationManager().requestWhenInUseAuthorization()
         CLLocationManager().desiredAccuracy = kCLLocationAccuracyBest
     }
+
+    func onButtonTapped() {
+        if let nearbyFarmer = findNearbyFarmer() {
+            mapCameraPosition = .region(MKCoordinateRegion(center: CLLocationCoordinate2D(latitude: nearbyFarmer.location.latitude, longitude: nearbyFarmer.location.longitude), span: MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)))
+        } else {
+            // TODO: request to the user to invite them to accept the location
+        }
+    }
+
+    private func findNearbyFarmer() -> Farmer? {
```

Comment:
You implementation contains side effects. Check on google what does it mean and if you still don't see what I mean let's discuss on slack quickly today ;)